spine_items/database_validation.py

```diff
@@ -0,0 +1,139 @@
+######################################################################################################################
+# Copyright (C) 2017-2022 Spine project consortium
+# This file is part of Spine Items.
+# Spine Items is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General
+# Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option)
+# any later version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General
+# Public License for more details. You should have received a copy of the GNU Lesser General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+######################################################################################################################
+"""Utilities to validate that a database exists."""
+from pathlib import Path
+
+from PySide6.QtCore import QObject, QRunnable, QThread, QThreadPool, QTimer, Signal, Slot
+from PySide6.QtWidgets import QApplication
+
+from spine_items.utils import check_database_url
+
+
+class _ValidationTask(QRunnable):
+    """Runnable that performs the actual validation in a worker thread."""
+
+    def __init__(self, dialect, sa_url, finish_slot, fail_slot, success_slot):
+        """
+        Args:
+            dialect (str): database dialect
+            sa_url (URL): SQLAlchemy URL
+            finish_slot (Callable): a Qt slot to connect to the finished signal
+            fail_slot (Callable): a Qt slot to connect to the fail signal
+            success_slot (Callable, optional): a Qt slot to connect to the success signal
+        """
+        super().__init__()
+        self._dialect = dialect
+        self._sa_url = sa_url
+        self._signals = _TaskSignals()
+        self._signals.validation_failed.connect(fail_slot)
+        if success_slot is not None:
+            self._signals.validation_succeeded.connect(success_slot)
+        self._signals.finished.connect(finish_slot)
+
+    def run(self):
+        """Checks if it is possible to establish an SQLAlchemy connection to the given URL."""
+        try:
+            self._signals.moveToThread(QThread.currentThread())
```

Comment:
I'm getting the below at loading a project that has Data Stores?
```
QObject::moveToThread: Current thread (0x5570c5237ce0) is not the object's thread (0x5570c4907db0).
Cannot move to target thread (0x5570c5237ce0)
```

Reply:
Interesting. I don't see this, could it be something Linux specific? In any case, I'll have a look.

Reply:
Haha, I should read the documentation more carefully:

> Warning: This function is not thread-safe; the current thread must be same as the current thread affinity. In other words, this function can only "push" an object from the current thread to another thread, it cannot "pull" an object from any arbitrary thread to the current thread. There is one exception to this rule however: objects with no thread affinity can be "pulled" to the current thread.

(from [Qt documentation](https://doc.qt.io/qt-6/qobject.html#moveToThread))

Sorry for the inconvenience! For some reason this worked perfectly fine on Windows. I'll try to get it fixed ASAP.

Reply:
@manuelma Could you try if `1668_moveToThread_issue` branch fixes the Data Store issue for you? I cannot reproduce the Traceback here. Meanwhile, I'll try to come up with a unit test that catches the Traceback in GitHub actions.